macros/base/homing/homing_overide.cfg

```diff
@@ -94,6 +96,22 @@ gcode:
             G0 Z{homing_zhop} F{z_drop_speed}
             {% set X, Y, Z = True, True, True %}
         {% endif %}
+    {% else %}
+        {% if ('z' in printer.toolhead.homed_axes) %}
+            {% if (printer.toolhead.position.z < homing_zhop) %}
+                {% if verbose %}
+                    { action_respond_info("Z too low, performing ZHOP") }
+                {% endif %}
+                G0 Z{homing_zhop} F{z_drop_speed}
+            {% endif %}
+        {% elif homing_force_zhop}
+            {% if verbose %}
+                { action_respond_info("Forcing z hop without homing z.") }
+            {% endif %}
+            SET_KINEMATIC_POSITION X=0 Y=0 Z=0
+            G0 Z{homing_zhop} F{z_drop_speed}
+            M84 # Motor off to "unhome" z for functional safety. This will also unhome all previously homed axis
```

Comment:
And on this part, I had a look at the documentation and it's perfectly safe to do: `SET_KINEMATIC_POSITION Z=0` without impacting X and Y.
So this means that you can remove the variable and the use of M84. Just do a `SET_KINEMATIC POSITION Z=0` and then the Z hop :)

Also please change the respond_info to : "Homing X or Y while Z is not homed, forcing ZHOP"